### DISCUSSION TITLE
Upgrade strategy for the self-hosted version (Docker)

### DISCUSSION THREAD

**Discussion by empz** [tolgee/tolgee-platform#2482] [Q&A] [answered]
I've just started with the self-hosted version using Docker and pining a specific tag. When you push new images to your Docker Hub, can I simply stop the app, change the tag number to a newer image and restart it? Do you offer backward compatibility with the previous version (or based in SemVer maybe?).

What to do if a major version with breaking changes is released? Would it be possible to upgrade seamlessly?

**Comment by JanCizmar** [comment, marked as answer]
Hey! Yeah, you can just change the version number to newer image and restart. Downgrading is not recommended, since database migrations are run, and there is no way how to run them reversly.

**Reply by mfrieling** [reply]
Additional question: How to do a full backup before upgrading and restore it in case the upgrade failed? Or in order to move the installation to a different host machine?

**Reply by marekbrze** [reply]
I am also interested in backups
